### PR TITLE
DA/group joining

### PR DESCRIPTION
**Pull request by d-atkins**
## Description
Completes [User Story 2: Joining a Group](https://app.asana.com/0/1170154762648575/1170576892702729/f)

## Notes
Creating an account by logging in with GitHub no longer sets a default group

Users belonging to groups is now optional

Added a form that allows a user to pick their group

Added a placeholder `Posts Index` for the sake of redirecting

Removed group association from `users_factory` - factory generated users now default to being groupless

## RSpec Results
```
26 examples, 0 failures

Coverage report generated for RSpec - 238 / 238 LOC (100.0%) covered.
```

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class PostsController < ApplicationController
```
Comment (d-atkins) [inline app/controllers/posts_controller.rb:1]: Added this only because joining a group redirects user back to posts index

File: app/controllers/profile/group_controller.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+class Profile::GroupController < ApplicationController
+  def new
+    @groups = Group.all
+  end
+
+  def create
+    group_id = group_params[:group_select]
+    group = Group.find_by(id: group_id)
+    if group
```
Comment (d-atkins) [inline app/controllers/profile/group_controller.rb:9]: this check is made incase a user edits their html and try to join an invalid group or a group is deleted before they are able to join one - will add actual tests for these edge cases later

File: app/controllers/profile/group_controller.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      current_user.update(group_id: group_id)
+      flash[:sucess] = "You have joined #{group.name}!"
+    end
+    redirect_to posts_path
+  end
+
+  private
+    def group_params
+      params.permit(:group_select)
```
Comment (d-atkins) [inline app/controllers/profile/group_controller.rb:18]: This param comes from the drop down box from the form and it holds the ID for the selected group
